make/test.mk

```diff
@@ -333,13 +333,16 @@ create-host-resources:
 	-oc create -f deploy/host-operator/${ENVIRONMENT}/ -n ${HOST_NS}
 	# patch toolchainconfig to prevent webhook deploy for 2nd member, a 2nd webhook deploy causes the webhook verification in e2e tests to fail
 	# since e2e environment has 2 member operators running in the same cluster
+	# for details on how the TOOLCHAINCLUSTER_NAME is composed see https://github.com/codeready-toolchain/toolchain-cicd/blob/master/scripts/add-cluster.sh
 	if [[ ${SECOND_MEMBER_MODE} == true ]]; then \
 		API_ENDPOINT=`oc get infrastructure cluster -o jsonpath='{.status.apiServerURL}'`; \
-		TOOLCHAIN_CLUSTER_NAME=`echo "$${API_ENDPOINT}" | sed 's/.*api\.\([^:]*\):.*/\1/'`; \
 		echo "API_ENDPOINT $${API_ENDPOINT}"; \
+		JOINING_CLUSTER_NAME=`echo "$${API_ENDPOINT}" | sed 's/^[^/]*\/\/\([^:/]*\)\(:.*\)\{0,1\}\(\/.*\)\{0,1\}$$/\1/' | sed 's/^api\.//'`; \
+		echo "Joining cluster name: $${JOINING_CLUSTER_NAME}"; \
+		TOOLCHAIN_CLUSTER_NAME=`echo "member-$${JOINING_CLUSTER_NAME:0:55}2"`; \
 		echo "TOOLCHAIN_CLUSTER_NAME $${TOOLCHAIN_CLUSTER_NAME}"; \
 		PATCH_FILE=/tmp/patch-toolchainconfig_${DATE_SUFFIX}.json; \
-		echo "{\"spec\":{\"members\":{\"specificPerMemberCluster\":{\"member-$${TOOLCHAIN_CLUSTER_NAME}2\":{\"webhook\":{\"deploy\":false}}}}}}" > $$PATCH_FILE; \
+		echo "{\"spec\":{\"members\":{\"specificPerMemberCluster\":{\"$${TOOLCHAIN_CLUSTER_NAME}\":{\"webhook\":{\"deploy\":false}}}}}}" > $$PATCH_FILE; \
```

Comment:
what if we swapped the targets in https://github.com/codeready-toolchain/toolchain-e2e/blob/7c7b289e8b9b2f7ec3c735493f7ce5363493d9dc/make/test.mk#L261 so the `create-host-resources` is called after `setup-toolchainclusters`
In that case, we wouldn't need to replicate the logic of getting the name, but instead of it we would:
1. list ToolchainCluster CRs of the member type with the role tenant
2. find the one which ends with `2`
3. take the name out of it

Reply:
Good point! Let me try that out !